graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/lookup/LookupAssignTtl.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.pipelineprocessor.functions.lookup;
+
+import com.google.inject.Inject;
+import org.graylog.plugins.pipelineprocessor.EvaluationContext;
+import org.graylog.plugins.pipelineprocessor.ast.functions.AbstractFunction;
+import org.graylog.plugins.pipelineprocessor.ast.functions.FunctionArgs;
+import org.graylog.plugins.pipelineprocessor.ast.functions.FunctionDescriptor;
+import org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor;
+import org.graylog2.lookup.LookupTableService;
+import org.graylog2.plugin.lookup.LookupResult;
+
+import java.util.List;
+
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.object;
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.string;
+
+public class LookupAssignTtl extends AbstractFunction<Object> {
+
+    public static final String NAME = "lookup_assign_ttl";
```

Comment:
Would a name like `lookup_set_value_with_ttl` follow our naming convention a little more? It does the same thing as `lookup_set_value` but just with the addition of a TTL. On that note, we also have a `lookup_set_string_list` pipeline function that we may also want to add a TTL to as well.

Reply:
With this pipeline function we're only setting the TTL for an existing entry with the specified key (see `MongoDBDataAdapterService.assignTtl())`, so I think the naming makes sense but open to changing it if you think otherwise.

Definitely agree that adding an optional parameter to `lookup_set_value` and `lookup_set_string_list` makes sense in the long run so opening a ticket for that as discussed in check-in. For now they could at least achieve the same thing by doing something like:
```
lookup_set_value("table", "key", "val");
lookup_assign_ttl("table", "key", 500);
```
Which I think is not too bad for the time being.